File: practicas/practica-3-en-raya/Juego.java
import java.util.Scanner; 

public class Juego {
    
    private Tablero tablero;
    private Jugador jugador1;
    private Jugador jugador2;
    private Jugador turnoActual;

    
    public Juego() {
        tablero = new Tablero();
        jugador1 = new Jugador("Jugador 1", 'X');
        jugador2 = new Jugador("Jugador 2", 'O');
        turnoActual = jugador1; 
    }

   
    public void iniciarPartida() {
        Scanner scanner = new Scanner(System.in);
        boolean juegoTerminado = false;

        System.out.println("¡Bienvenido al Juego de 3 en Raya! por: Alin Anabi");

        
        while (!juegoTerminado) {
            tablero.mostrarTablero();
            System.out.println("Turno de " + turnoActual.getNombre() + " (" + turnoActual.getFicha() + ")");
            
            
            System.out.print("Ingresa la fila (0, 1 o 2): ");
            int fila = scanner.nextInt();
            System.out.print("Ingresa la columna (0, 1 o 2): ");
            int columna = scanner.nextInt();

            
            if (tablero.colocarFicha(fila, columna, turnoActual.getFicha())) {
                
                
                if (tablero.verificarGanador(turnoActual.getFicha())) {
                    tablero.mostrarTablero();
                    System.out.println("¡FELICIDADES! Ha ganado el " + turnoActual.getNombre());
                    juegoTerminado = true; 
                } 
                
                else if (tablero.tableroLleno()) {
                    tablero.mostrarTablero();
                    System.out.println("¡Es un empate! El tablero está lleno.");
                    juegoTerminado = true; 
                } 
                
                else {
                    cambiarTurno();
                }
                
            } else {
                
                System.out.println("Movimiento inválido. Revisa que la casilla esté vacía y los números sean del 0 al 2.");
            }
        }
        
        scanner.close(); 
    }

    
    private void cambiarTurno() {
        if (turnoActual == jugador1) {
            turnoActual = jugador2;
        } else {
            turnoActual = jugador1;
        }
    }
}

File: 3 en raya en c#/Services/Juego.cs
using System;
using TresEnRaya.Models;

namespace TresEnRaya.Services
{
    public class Juego
    {
        private Jugador jugador1;
        private Jugador jugador2;
        private Jugador jugadorActual;
        private Tablero tablero;

        public Juego()
        {
            tablero = new Tablero();
            ConfigurarJugadores();
        }

        private void ConfigurarJugadores()
        {
            Console.WriteLine("=== CONFIGURACIÓN DEL JUEGO 3 EN RAYA ===");
            Console.Write("Ingrese el nombre del Jugador 1 (Ficha X): ");
            string nombre1 = Console.ReadLine() ?? "Jugador 1";
            
            Console.Write("Ingrese el nombre del Jugador 2 (Ficha O): ");
            string nombre2 = Console.ReadLine() ?? "Jugador 2";

            jugador1 = new Jugador(nombre1, 'X');
            jugador2 = new Jugador(nombre2, 'O');
            jugadorActual = jugador1; 
        }

        private void CambiarTurno()
        {
            jugadorActual = (jugadorActual == jugador1) ? jugador2 : jugador1;
        }

        public void IniciarPartida()
        {
            bool juegoTerminado = false;

            while (!juegoTerminado)
            {
                Console.Clear();
                Console.WriteLine($"Turno actual: {jugadorActual.Nombre} ({jugadorActual.Simbolo})");
                tablero.MostrarTablero();

                Console.WriteLine("Ingrese su movimiento:");
                Console.Write("Fila (0-2): ");
                bool entradaValidaFila = int.TryParse(Console.ReadLine(), out int fila);
                Console.Write("Columna (0-2): ");
                bool entradaValidaColumna = int.TryParse(Console.ReadLine(), out int columna);

                if (!entradaValidaFila || !entradaValidaColumna)
                {
                    Console.WriteLine("Por favor, ingrese números válidos.");
                    System.Threading.Thread.Sleep(1500);
                    continue;
                }

                bool movimientoExitoso = tablero.ColocarFicha(fila, columna, jugadorActual.Simbolo);
                if (!movimientoExitoso)
                {
                    System.Threading.Thread.Sleep(1500);
                    continue;
                }

                if (tablero.VerificarGanador(jugadorActual.Simbolo))
                {
                    Console.Clear();
                    tablero.MostrarTablero();
                    Console.WriteLine($"¡Felicidades! {jugadorActual.Nombre} ha ganado la partida.");
                    juegoTerminado = true;
                }
                else if (tablero.EstaLleno())
                {
                    Console.Clear();
                    tablero.MostrarTablero();
                    Console.WriteLine("¡Empate! El tablero está completo y no hay ganador.");
                    juegoTerminado = true;
                }
                else
                {
                    CambiarTurno();
                }
            }
        }
    }
}